### PR TITLE
fix(ci): further consolidate NODE_OPTIONS

### PR DESCRIPTION
**Pull request by david-leifker**
Prevent having to track down and modify memory settings in multiple places for NODE_OPTIONS

@chakru-r when trying to enable coverage, I think we're missing tests? Disabling for now.

```
$ NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' vitest run --coverage run
vs  node_modules/monaco-editor/min-maps/vs/

 RUN  v0.34.6 /Users/dleifker/Code/datahub/datahub-web-react
      Coverage enabled with v8

 % Coverage report from v8
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |                   
----------|---------|----------|---------|---------|-------------------
filter:  run
include: **/*.{test,spec}.?(c|m)[jt]s?(x)
exclude:  **/node_modules/**, **/dist/**, **/cypress/**, **/.{idea,git,cache,output,temp}/**, **/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*
watch exclude:  **/node_modules/**, **/dist/**

No test files found, exiting with code 1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
[Incubating] Problems report is available at: file:///Users/dleifker/Code/datahub/build/reports/problems/problems-report.html

```

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/package.json [hunk 1]
```diff
@@ -87,20 +87,20 @@
...
-        "analyze": "source-map-explorer 'dist/assets/*.js'",
-        "start": "yarn run generate && vite",
-        "ec2-dev": "yarn run generate && CI=true;export CI;vite",
-        "build": "yarn run generate && NODE_OPTIONS='--max-old-space-size=4096 --openssl-legacy-provider' CI=false vite build",
-        "test": "vitest",
-        "generate": "graphql-codegen --config codegen.yml",
+        "analyze": "NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' source-map-explorer 'dist/assets/*.js'",
+        "start": "NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' yarn run generate && NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' vite",
+        "ec2-dev": "NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' yarn run generate && CI=true NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' vite",
+        "build": "NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' yarn run generate && CI=false NODE_OPTIONS='--max-old-space-size=5120 --openssl-legacy-provider' vite build",
```
Comment (hsheth2) [inline datahub-web-react/package.json:93]: this should only be necessary on the `vite build` step

**Review by hsheth2 — APPROVED** [review]

File: datahub-web-react/package.json (hunk 1 above)
Comment (david-leifker) [reply datahub-web-react/package.json]: Removed it from several of them. The storybook build causes OOM in Saas. Enabling coverage on the test also triggers OOM so more then just `vite build` is likely warranted.

File: datahub-web-react/build.gradle [hunk 2]
```diff
@@ -79,7 +79,7 @@ task yarnServe(type: YarnTask, dependsOn: [yarnInstall, yarnGenerate]) {
 
 task yarnTest(type: YarnTask, dependsOn: [yarnInstall, yarnGenerate]) {
   // Explicitly runs in non-watch mode.
-  args = ['run', 'test', 'run']
+  args = ['run', project.hasProperty('withCoverage') ? 'test-coverage' : 'test', 'run']
```
Comment (chakru-r) [inline datahub-web-react/build.gradle:82]: Where is this property being set? We would want the coverage to be generated when gradle runs the `test` task (like all other projects).

Comment (david-leifker) [reply datahub-web-react/build.gradle:82]: Good catch, should be added to the frontend build in the build-and-test workflow but in general doesn't need to be run in other places I think. 
`name: Gradle build (and test) for frontend`

**Pull request merged**